OpenSim/Common/DataTable.h

```diff
@@ -375,19 +380,19 @@ class DataTable_ : public AbstractDataTable {
         }
         setColumnLabels(thisLabels);
 
-        // Form rows for this table from that table.
+        // Construct matrix for this table from that table.
+        _depData.resize((int)that.getNumRows(), 
+            (int)that.getNumColumns() / numComponentsPerElement());
         for(unsigned r = 0; r < that.getNumRows(); ++r) {
-            const auto& thatInd = that.getIndependentColumn().at(r);
             auto thatRow = that.getRowAtIndex(r).getAsRowVector();
-            std::vector<ETY> thisRow{};
             for(unsigned c = 0;
-                c < that.getNumColumns();
-                c += numComponentsPerElement()) {
-                thisRow.push_back(makeElement(thatRow.begin() + c,
-                                              thatRow.end()));
+                c < that.getNumColumns() / numComponentsPerElement(); ++c) {
```

Comment:
Does it make sense to have a local variable for the new number of columns?

Reply:
Since `_depData` has already been resized, I could use `this->getNumColumns()`.

Reply:
+1